### PR TITLE
Fix rendering emails with <10 items as table with trace logging

### LINKED ISSUE #21166: Send email fails when <10 items rendered as table and custom (verbose) logging level
**Describe the bug**
Interesting bug that happens when combining:
- TRACE logging level in Log4j2 (does not happen with lower level) on pulses and emails
- a dashboard subscription that has any viz that will be rendered in the subscription as a table (table, map) with less than 10 items

**Logs**
<details>

```
metabase_cus_log    | 2022-03-22 19:00:40,765 DEBUG metabase.pulse :: Sending Pulse (1: "b") with 1 Cards via email
metabase_cus_log    | 2022-03-22 19:00:40,769 TRACE pulse.render :: Detected chart type :table for Card "People, Count, Grouped by City" because no other chart types match
metabase_cus_log    | 2022-03-22 19:00:40,770 DEBUG pulse.render :: Rendering pulse card with chart-type :table and render-type :attachment
metabase_cus_log    | 2022-03-22 19:00:40,786 TRACE pulse.render :: Detected chart type :table for Card "People, Count, Grouped by City" because no other chart types match
metabase_cus_log    | 2022-03-22 19:00:40,908 ERROR middleware.log :: POST /api/pulse/test 500 67.6 ms (32 DB calls) 
metabase_cus_log    | {:via
metabase_cus_log    |  [{:type java.util.IllegalFormatConversionException,
metabase_cus_log    |    :message "d != java.lang.Double",
metabase_cus_log    |    :at [java.util.Formatter$FormatSpecifier failConversion nil -1]}],
metabase_cus_log    |  :trace
metabase_cus_log    |  [[java.util.Formatter$FormatSpecifier failConversion nil -1]
metabase_cus_log    |   [java.util.Formatter$FormatSpecifier printInteger nil -1]
metabase_cus_log    |   [java.util.Formatter$FormatSpecifier print nil -1]
metabase_cus_log    |   [java.util.Formatter format nil -1]
metabase_cus_log    |   [java.util.Formatter format nil -1]
metabase_cus_log    |   [java.lang.String format nil -1]
metabase_cus_log    |   [clojure.core$format invokeStatic "core.clj" 5733]
metabase_cus_log    |   [clojure.core$format doInvoke "core.clj" 5727]
metabase_cus_log    |   [clojure.lang.RestFn applyTo "RestFn.java" 139]
metabase_cus_log    |   [clojure.core$apply invokeStatic "core.clj" 669]
metabase_cus_log    |   [clojure.core$apply invoke "core.clj" 662]
metabase_cus_log    |   [metabase.email.messages$include_csv_attachment_QMARK_$no__59796 doInvoke "messages.clj" 336]
metabase_cus_log    |   [clojure.lang.RestFn invoke "RestFn.java" 439]
metabase_cus_log    |   [metabase.email.messages$include_csv_attachment_QMARK_ invokeStatic "messages.clj" 358]
metabase_cus_log    |   [metabase.email.messages$include_csv_attachment_QMARK_ invoke "messages.clj" 329]
metabase_cus_log    |   [metabase.email.messages$result_attachment invokeStatic "messages.clj" 394]
metabase_cus_log    |   [metabase.email.messages$result_attachment invoke "messages.clj" 391]
metabase_cus_log    |   [clojure.core$map$fn__5884 invoke "core.clj" 2759]
metabase_cus_log    |   [clojure.lang.LazySeq sval "LazySeq.java" 42]
metabase_cus_log    |   [clojure.lang.LazySeq seq "LazySeq.java" 51]
metabase_cus_log    |   [clojure.lang.RT seq "RT.java" 535]
metabase_cus_log    |   [clojure.core$seq__5419 invokeStatic "core.clj" 139]
metabase_cus_log    |   [clojure.core$apply invokeStatic "core.clj" 662]
metabase_cus_log    |   [clojure.core$mapcat invokeStatic "core.clj" 2787]
metabase_cus_log    |   [clojure.core$mapcat doInvoke "core.clj" 2787]
metabase_cus_log    |   [clojure.lang.RestFn invoke "RestFn.java" 423]
metabase_cus_log    |   [metabase.email.messages$result_attachments invokeStatic "messages.clj" 406]
metabase_cus_log    |   [metabase.email.messages$result_attachments invoke "messages.clj" 405]
metabase_cus_log    |   [metabase.email.messages$render_message_body invokeStatic "messages.clj" 473]
metabase_cus_log    |   [metabase.email.messages$render_message_body invoke "messages.clj" 456]
metabase_cus_log    |   [metabase.email.messages$render_pulse_email invokeStatic "messages.clj" 485]
metabase_cus_log    |   [metabase.email.messages$render_pulse_email invoke "messages.clj" 482]
metabase_cus_log    |   [metabase.pulse$fn__75689 invokeStatic "pulse.clj" 293]
metabase_cus_log    |   [metabase.pulse$fn__75689 invoke "pulse.clj" 282]
metabase_cus_log    |   [clojure.lang.MultiFn invoke "MultiFn.java" 239]
metabase_cus_log    |   [metabase.pulse$results__GT_notifications$iter__75719__75723$fn__75724 invoke "pulse.clj" 345]
metabase_cus_log    |   [clojure.lang.LazySeq sval "LazySeq.java" 42]
metabase_cus_log    |   [clojure.lang.LazySeq seq "LazySeq.java" 51]
metabase_cus_log    |   [clojure.lang.RT seq "RT.java" 535]
metabase_cus_log    |   [clojure.core$seq__5419 invokeStatic "core.clj" 139]
metabase_cus_log    |   [clojure.core$seq__5419 invoke "core.clj" 139]
metabase_cus_log    |   [metabase.pulse$send_notifications_BANG_ invokeStatic "pulse.clj" 387]
metabase_cus_log    |   [metabase.pulse$send_notifications_BANG_ invoke "pulse.clj" 386]
metabase_cus_log    |   [metabase.pulse$send_pulse_BANG_ invokeStatic "pulse.clj" 415]
metabase_cus_log    |   [metabase.pulse$send_pulse_BANG_ doInvoke "pulse.clj" 395]
metabase_cus_log    |   [clojure.lang.RestFn invoke "RestFn.java" 410]
metabase_cus_log    |   [metabase.api.pulse$fn__76683 invokeStatic "pulse.clj" 211]
metabase_cus_log    |   [metabase.api.pulse$fn__76683 invoke "pulse.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_response$fn__32279 invoke "core.clj" 160]
metabase_cus_log    |   [compojure.core$wrap_route_middleware$fn__32263 invoke "core.clj" 132]
metabase_cus_log    |   [compojure.core$wrap_route_info$fn__32268 invoke "core.clj" 139]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 151]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 152]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [metabase.server.middleware.auth$enforce_authentication$fn__66100 invoke "auth.clj" 14]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [compojure.core$make_context$handler__32319 invoke "core.clj" 289]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 299]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 199]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 199]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 199]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [metabase.api.routes$fn__77241$fn__77242 invoke "routes.clj" 54]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [clojure.lang.AFn applyToHelper "AFn.java" 160]
metabase_cus_log    |   [clojure.lang.AFn applyTo "AFn.java" 144]
metabase_cus_log    |   [clojure.core$apply invokeStatic "core.clj" 667]
metabase_cus_log    |   [clojure.core$apply invoke "core.clj" 662]
metabase_cus_log    |   [metabase.server.routes$fn__77386$fn__77387 doInvoke "routes.clj" 57]
metabase_cus_log    |   [clojure.lang.RestFn invoke "RestFn.java" 436]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [compojure.core$make_context$handler__32319 invoke "core.clj" 289]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 299]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$wrap_route_matches$fn__32272 invoke "core.clj" 153]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 199]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292$respond_SINGLEQUOTE___32293 invoke "core.clj" 197]
metabase_cus_log    |   [compojure.core$make_context$fn__32323 invoke "core.clj" 300]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [compojure.core$routes$fn__32291$f__32292 invoke "core.clj" 198]
metabase_cus_log    |   [compojure.core$routes$fn__32291 invoke "core.clj" 200]
metabase_cus_log    |   [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__74369 invoke "exceptions.clj" 98]
metabase_cus_log    |   [metabase.server.middleware.exceptions$catch_api_exceptions$fn__74366 invoke "exceptions.clj" 86]
metabase_cus_log    |   [metabase.server.middleware.log$log_api_call$fn__79450$fn__79451$fn__79452 invoke "log.clj" 209]
metabase_cus_log    |   [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic "diagnostic.clj" 15]
metabase_cus_log    |   [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke "diagnostic.clj" 9]
metabase_cus_log    |   [metabase.server.middleware.log$log_api_call$fn__79450$fn__79451 invoke "log.clj" 201]
metabase_cus_log    |   [toucan.db$_do_with_call_counting invokeStatic "db.clj" 216]
metabase_cus_log    |   [toucan.db$_do_with_call_counting invoke "db.clj" 209]
metabase_cus_log    |   [metabase.server.middleware.log$log_api_call$fn__79450 invoke "log.clj" 200]
metabase_cus_log    |   [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__84842 invoke "browser_cookie.clj" 38]
metabase_cus_log    |   [metabase.server.middleware.security$add_security_headers$fn__56115 invoke "security.clj" 148]
metabase_cus_log    |   [metabase.server.middleware.json$wrap_json_body$fn__82154 invoke "json.clj" 62]
metabase_cus_log    |   [metabase.server.middleware.json$wrap_streamed_json_response$fn__82172 invoke "json.clj" 98]
metabase_cus_log    |   [metabase.server.middleware.offset_paging$handle_paging$fn__56139 invoke "offset_paging.clj" 42]
metabase_cus_log    |   [ring.middleware.keyword_params$wrap_keyword_params$fn__85109 invoke "keyword_params.clj" 55]
metabase_cus_log    |   [ring.middleware.params$wrap_params$fn__85125 invoke "params.clj" 69]
metabase_cus_log    |   [metabase.server.middleware.misc$maybe_set_site_url$fn__34385 invoke "misc.clj" 59]
metabase_cus_log    |   [metabase.server.middleware.session$bind_current_user$fn__46674$fn__46675 invoke "session.clj" 262]
metabase_cus_log    |   [metabase.server.middleware.session$do_with_current_user invokeStatic "session.clj" 243]
metabase_cus_log    |   [metabase.server.middleware.session$do_with_current_user invoke "session.clj" 235]
metabase_cus_log    |   [metabase.server.middleware.session$bind_current_user$fn__46674 invoke "session.clj" 261]
metabase_cus_log    |   [metabase.server.middleware.session$wrap_current_user_info$fn__46661 invoke "session.clj" 221]
metabase_cus_log    |   [metabase.server.middleware.session$wrap_session_id$fn__46647 invoke "session.clj" 167]
metabase_cus_log    |   [metabase.server.middleware.auth$wrap_api_key$fn__66108 invoke "auth.clj" 27]
metabase_cus_log    |   [ring.middleware.cookies$wrap_cookies$fn__85029 invoke "cookies.clj" 216]
metabase_cus_log    |   [metabase.server.middleware.misc$add_content_type$fn__34368 invoke "misc.clj" 27]
metabase_cus_log    |   [metabase.server.middleware.misc$disable_streaming_buffering$fn__34393 invoke "misc.clj" 76]
metabase_cus_log    |   [ring.middleware.gzip$wrap_gzip$fn__85071 invoke "gzip.clj" 86]
metabase_cus_log    |   [metabase.server.middleware.misc$bind_request$fn__34396 invoke "misc.clj" 93]
metabase_cus_log    |   [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__84858 invoke "ssl.clj" 38]
metabase_cus_log    |   [metabase.server$async_proxy_handler$fn__79223 invoke "server.clj" 73]
metabase_cus_log    |   [metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a handle nil -1]
metabase_cus_log    |   [org.eclipse.jetty.server.handler.HandlerWrapper handle "HandlerWrapper.java" 127]
metabase_cus_log    |   [org.eclipse.jetty.server.Server handle "Server.java" 516]
metabase_cus_log    |   [org.eclipse.jetty.server.HttpChannel lambda$handle$1 "HttpChannel.java" 388]
metabase_cus_log    |   [org.eclipse.jetty.server.HttpChannel dispatch "HttpChannel.java" 633]
metabase_cus_log    |   [org.eclipse.jetty.server.HttpChannel handle "HttpChannel.java" 380]
metabase_cus_log    |   [org.eclipse.jetty.server.HttpConnection onFillable "HttpConnection.java" 277]
metabase_cus_log    |   [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded "AbstractConnection.java" 311]
metabase_cus_log    |   [org.eclipse.jetty.io.FillInterest fillable "FillInterest.java" 105]
metabase_cus_log    |   [org.eclipse.jetty.io.ChannelEndPoint$1 run "ChannelEndPoint.java" 104]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.strategy.EatWhatYouKill runTask "EatWhatYouKill.java" 338]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.strategy.EatWhatYouKill doProduce "EatWhatYouKill.java" 315]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.strategy.EatWhatYouKill tryProduce "EatWhatYouKill.java" 173]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.strategy.EatWhatYouKill run "EatWhatYouKill.java" 131]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread run "ReservedThreadExecutor.java" 386]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.QueuedThreadPool runJob "QueuedThreadPool.java" 883]
metabase_cus_log    |   [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run "QueuedThreadPool.java" 1034]
metabase_cus_log    |   [java.lang.Thread run nil -1]],
metabase_cus_log    |  :cause "d != java.lang.Double",
metabase_cus_log    |  :message "d != java.lang.Double"}
```

</details>

**To Reproduce**
1) configure this tracing level https://log4j.us/v2/templates/metabase?trace=metabase.task,metabase.pulse,metabase.email
2) start metabase (with maildev configured) and do a question with less than 10 rows, add it to a dashboard
3) try doing a subscription and clicking on send email now, see the error

**Expected behavior**
Mail to send

**Screenshots**
NA

**Information about your Metabase Installation:**
- Your browser and the version: Brave latest
- Your operating system: Ubuntu LTS
- Your databases: H2
- Metabase version: 42.2
- Metabase hosting environment: Docker
- Metabase internal database: H2

**Severity**
Not severe (unless you really need the trace level)

**Additional context**
If you do the question with 10 rows, this does not happen. Also if you raise the log4j2 level, this also doesn't happen. If you use any viz that is not rendered as a table, this also doesn't happen

### PR DESCRIPTION
**Pull request by metamben**
Fixes #21166.

### CODE REVIEW

**Comment by metamben** [comment]
I've added a test, but I don't think it will be useful in the future. It's testing a very specific edge case which is highly unlikely to happen again. Should we remove it?

File: test/metabase/email/messages_test.clj [hunk 1]
```diff
@@ -81,3 +81,20 @@
...
+(deftest render-pulse-email-test
+  (testing "Email with few rows and columns can be rendered when tracing"
+    (let [logger-name (namespace ::messages/whatever)
+          logger (org.apache.logging.log4j.LogManager/getLogger logger-name)
+          log-level (.getLevel logger)
+          result {:card {:name "card-name"}
+                  :result {:data {:cols [{:name "x"} {:name "y"}]
+                                  :rows [[0 0]
+                                         [1 1]]}}}]
+      (.setLevel logger org.apache.logging.log4j.Level/TRACE)
```
Comment (qnkhuat) [inline test/metabase/email/messages_test.clj:94]: I think you could use `mt/with-log-level` to temporarily change the log level for this test.

File: src/metabase/email/messages.clj [hunk 2]
```diff
@@ -333,14 +333,11 @@
       (not= :table (render/detect-pulse-chart-type card nil result-data))
       (no "we've determined it should not be rendered as a table")
 
-      (= (count (take render.body/cols-limit cols)) render.body/cols-limit)
```
Comment (qnkhuat) [inline src/metabase/email/messages.clj:336]: If I understand correctly `cols-limit` always is `##INF` so think it's okay to remove this as a temporarily fix , but wdyt we remove this `cols-limit` variable all together?
Cam has a TODO about this https://github.com/metabase/metabase/blob/91bd47814a4f07a5aa2df589b91492d56f92fa7b/src/metabase/pulse/render/body.clj#L58 and I think it's safe to remove it now.

**Review by qnkhuat — APPROVED** [review]
Your fix works, so approving. But lmk if my comment makes sense.

File: test/metabase/email/messages_test.clj (hunk 1 above)
Comment (metamben) [reply test/metabase/email/messages_test.clj:94]: Right, thanks for the hint! On the other hand, as I wrote in my comment, I'm not sure at all it makes much sense having this test. What do you think?

**Review by noahmoss — APPROVED** [review]

**Pull request merged**